File: TheaterSuggestions/CSharp/TheaterSuggestions.Tests/Row.cs
using System.Collections.Generic;

namespace SeatsSuggestions.Tests
{
    public class Row
    {
        public string Name { get; set; }
        public List<Seat> Seats { get; set; } = new List<Seat>();

        public SeatAllocation FindAllocation(int partyRequested, PricingCategory pricingCategory)
        {
            foreach (var seat in Seats)
            {
                if (seat.IsAvailable() && seat.MatchCategory(pricingCategory))
                {
                    var seatAllocation = new SeatAllocation(partyRequested, pricingCategory);
                    seatAllocation.AddSeat(seat);

                    if (seatAllocation.MatchExpectation())
                    {
                        return seatAllocation;
                    }
                }
            }

            return new AllocationNotAvailable(partyRequested, pricingCategory);
        }
    }
}

File: TheaterSuggestions/Java/SeatsSuggestionsAcceptanceTests/src/test/java/com/baasie/SeatsSuggestionsAcceptanceTests/Row.java
package com.baasie.SeatsSuggestionsAcceptanceTests;

import java.util.ArrayList;
import java.util.List;

public class Row {
    private String name;
    private List<Seat> seats = new ArrayList<>();

    public List<Seat> seats() {
        return seats;
    }
}
